scripts/cfengine-build-host-setup.cf

```diff
@@ -0,0 +1,205 @@
+body file control
+{
+  inputs => { "$(sys.libdir)/stdlib.cf" };
+}
+# assumes that the operating system has working repo lists and has been updated/upgraded already
+
+bundle agent cfengine_build_host_setup
+{
+  packages:
+    debian_9|debian_10::
+      "python-psycopg2";
+    debian_11|debian_12::
+      "python3-psycopg2";
+    ubuntu_16:: # ubuntu_16 doesn't have systemd-coredump by default?
+      "systemd-coredump";
+    ubuntu_20::
+      "autoconf"; # because on arm ubu20 we need to reconfigure
+      "shellcheck"; # not sure why, ubu20 addition
+    debian.(!debian_12.!ubuntu_22)::
+      "python"; # debian_12 has only python3
+     !(debian_9|ubuntu_16).(debian|ubuntu)::
+       "default-jre"; # on debian10+ and ubuntu18+ this will be jdk11, good enough for jenkins 2.426.1 https://www.jenkins.io/doc/book/platform-information/support-policy-java/index.html
+
+# these packages were apt purge in shell scripts, package_method generic should do the same thing with package_method => "delete"
+    debian|ubuntu::
+      "libltdl7" package_policy => "delete";
+      "libltdl-dev" package_policy => "delete";
+
+# todo --allow-change-held-packages introduced in debian-12, why?
```

Comment:
What is this comment about? Us introducing that option to something with starting to support debian 12 or debian 12 got that new option?

--allow-change-held-packages
           Force yes; this is a dangerous option that will cause apt to continue without
           prompting if it is changing held packages. It should not be used except in very
           special situations. Using it can potentially destroy your system! Configuration Item:
           APT::Get::allow-change-held-packages. Introduced in APT 1.1.
           
`apt-mark showhold` should list them.

Reply:
right, a comment about stuff in the source documents like jenkins-jobs/config.xml aka cloud configs and such.